pkgs/development/libraries/phash/default.nix

```diff
@@ -24,6 +24,15 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-frISiZ89ei7XfI5F2nJJehfQZsk0Mlb4n91q/AiZ2vA=";
   };
 
+  NIX_LDFLAGS = "-lfftw3_threads";
+
+  patches = [
+    (fetchpatch {
```

Comment:
1. Please add comment like `# Add proper pthread return value to fix build`
2. Seems like this is not merged, better to put patch file here because you may be required to force push

Reply:
1. done
1. could you please elaborate on that one? Both approaches, adding the patch in the repo and referring to it via the URl, would be referring to the same patch even if I was required to force push upstream. Other than GitHub potentially GC'ing the corresponding patch eventually if I were to force push and then let the repo just sit there for years I don't see anything that'd be problematic and even in that case the nixpkgs infrastructure (binary cache and hydra) would keep the patch around, being able to reference it based on the provided hash alone, which means worst case someone would still be able to grab that patch file years later and shove it into nixpkgs anyway. Maybe referencing to the commit specifically in my fork of the repo would be better though since then the link to the commit (i.e. when removing the *.patch* suffix) shows what PR the commit belongs to (kind of odd that GitHub doesn't show that when referencing the commit in the base repo despite there being a PR). I've just assumed referencing patches externally would be preferred similarly to how *Cargo.lock* files in nixpkgs are an issue bloating its size significantly when diffs are taken into account (which is less of a problem here but just comparing the ~100 bytes for URl and hash with the full 1.5kB patch is still an order of magnitude).

Reply:
See https://github.com/NixOS/nixpkgs/tree/master/pkgs#patches.

Reply:
Thank you very much, I didn't know that documentation existed.
The patch has been exported using `git format-patch` and added to nixpkgs.
I've also rebased, resolved the merge conflict, and tested whether the patch still works, everything should be fine with the latest push.